### PR TITLE
[LG-2858] Aria label props

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Exports `AriaLabelProps` from `a11y`, a union interface requiring _either_ `aria-label` or `aria-labelledby`

Used in `SearchInput` PR: https://github.com/mongodb/leafygreen-ui/pull/1528

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/a11y/src/AriaLabelProps.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * A union interface requiring _either_ `aria-label` or `aria-labelledby`
+ */
+export type AriaLabelProps =
```
Comment (shaneeza) [inline packages/a11y/src/AriaLabelProps.ts:4]: Would this only be for a component that doesn't accept the `label` prop?

Comment (TheSonOfThomp) [reply packages/a11y/src/AriaLabelProps.ts:4]: Yeah, in that case you'd extend this interface

Comment (shaneeza) [reply packages/a11y/src/AriaLabelProps.ts:4]: maybe we should also expose an interface that uses `Pick` for when the three are optional

**Review by shaneeza — APPROVED** [review]

File: packages/a11y/src/AriaLabelProps.ts (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/a11y/src/AriaLabelProps.ts:4]: Added `AriaLabelPropsWithLabel`

Comment (shaneeza) [reply packages/a11y/src/AriaLabelProps.ts:4]: oops, meant to say `Either` but this also works!

**Review by shaneeza — APPROVED** [review]

File: packages/a11y/src/AriaLabelProps.ts (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/a11y/src/AriaLabelProps.ts:4]: I think Either only works with 2 keys. It's easier to export this from here anyway I think

File: packages/a11y/src/AriaLabelProps.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * A union interface requiring _either_ `label`, `aria-label` or `aria-labelledby`
+ */
+export type AriaLabelPropsWithLabel =
+  | ({
+      label?: string;
+    } & AriaLabelProps)
+  | ({
+      label: string;
+    } & Partial<AriaLabelProps>);
```
Comment (shaneeza) [inline packages/a11y/src/AriaLabelProps.ts:23]: oh nice approach

**Pull request merged**